src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -264,204 +247,52 @@ namespace InternalHeatGains {
-        MaxAlpha = max(MaxAlpha, NumAlpha);
-        MaxNumber = max(MaxNumber, NumNumber);
-        CurrentModuleObject = "Lights";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, CurrentModuleObject, Loop, NumAlpha, NumNumber);
-        MaxAlpha = max(MaxAlpha, NumAlpha);
-        MaxNumber = max(MaxNumber, NumNumber);
-        CurrentModuleObject = "ElectricEquipment";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, CurrentModuleObject, Loop, NumAlpha, NumNumber);
-        MaxAlpha = max(MaxAlpha, NumAlpha);
-        MaxNumber = max(MaxNumber, NumNumber);
-        CurrentModuleObject = "GasEquipment";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, CurrentModuleObject, Loop, NumAlpha, NumNumber);
-        MaxAlpha = max(MaxAlpha, NumAlpha);
-        MaxNumber = max(MaxNumber, NumNumber);
-        CurrentModuleObject = "HotWaterEquipment";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, CurrentModuleObject, Loop, NumAlpha, NumNumber);
-        MaxAlpha = max(MaxAlpha, NumAlpha);
-        MaxNumber = max(MaxNumber, NumNumber);
-        CurrentModuleObject = "SteamEquipment";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, CurrentModuleObject, Loop, NumAlpha, NumNumber);
-        MaxAlpha = max(MaxAlpha, NumAlpha);
-        MaxNumber = max(MaxNumber, NumNumber);
-        CurrentModuleObject = "OtherEquipment";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, CurrentModuleObject, Loop, NumAlpha, NumNumber);
-        MaxAlpha = max(MaxAlpha, NumAlpha);
-        MaxNumber = max(MaxNumber, NumNumber);
-        CurrentModuleObject = "ElectricEquipment:ITE:AirCooled";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, CurrentModuleObject, Loop, NumAlpha, NumNumber);
-        MaxAlpha = max(MaxAlpha, NumAlpha);
-        MaxNumber = max(MaxNumber, NumNumber);
-        CurrentModuleObject = "ZoneBaseboard:OutdoorTemperatureControlled";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, CurrentModuleObject, Loop, NumAlpha, NumNumber);
-        MaxAlpha = max(MaxAlpha, NumAlpha);
-        MaxNumber = max(MaxNumber, NumNumber);
-        CurrentModuleObject = "ZoneContaminantSourceAndSink:CarbonDioxide";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, CurrentModuleObject, Loop, NumAlpha, NumNumber);
-        MaxAlpha = max(MaxAlpha, NumAlpha);
-        MaxNumber = max(MaxNumber, NumNumber);
-
-        IHGNumbers.allocate(MaxNumber);
-        AlphaName.allocate(MaxAlpha);
-        IHGNumbers = 0.0;
-        AlphaName = "";
-
-        // CurrentModuleObject='Zone'
-        for (Loop = 1; Loop <= state.dataGlobal->NumOfZones; ++Loop) {
-            // Overall Zone Variables
-            SetupOutputVariable(state,
```

Comment:
All of the `SetupOutputVariable` calls for the internal gains that changed are moved to a new function`setupIHGOutputs`.